tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/OpenIdTests.java

```diff
@@ -0,0 +1,210 @@
+import com.gargoylesoftware.htmlunit.html.HtmlForm;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+import com.gargoylesoftware.htmlunit.html.HtmlSubmitInput;
+import dasniko.testcontainers.keycloak.KeycloakContainer;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.tests.testers.JettyHomeTester;
+import org.eclipse.jetty.tests.testers.Tester;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.keycloak.admin.client.CreatedResponseUtil;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.representations.idm.CredentialRepresentation;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class OpenIdTests extends AbstractJettyHomeTest
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(OpenIdTests.class);
+
+    private static final KeycloakContainer KEYCLOAK_CONTAINER = new KeycloakContainer();
+
+    private static final String clientId = "jetty-api";
+    private static final String clientSecret = "JettyRocks!";
+
+    private static final String userName = "jetty";
+    private static final String password = "JettyRocks!Really";
+
+    private static final String firstName = "John";
+    private static final String lastName = "Doe";
+    private static final String email = "jetty@jetty.org";
+
+    private static String userId;
+
+    @BeforeAll
+    public static void startKeycloak()
+    {
+        KEYCLOAK_CONTAINER.start();
+        // init keycloak
+        try (Keycloak keycloak = KEYCLOAK_CONTAINER.getKeycloakAdminClient())
+        {
+            RealmRepresentation jettyRealm = new RealmRepresentation();
+            jettyRealm.setId("jetty");
+            jettyRealm.setRealm("jetty");
+            jettyRealm.setEnabled(true);
+            keycloak.realms().create(jettyRealm);
+
+            ClientRepresentation clientRepresentation = new ClientRepresentation();
+            clientRepresentation.setId("jetty-api");
+            clientRepresentation.setClientId("jetty-api");
+            clientRepresentation.setSecret(clientSecret);
+            clientRepresentation.setRedirectUris(List.of("http://localhost:*"));
+            clientRepresentation.setEnabled(true);
+            clientRepresentation.setPublicClient(Boolean.TRUE);
```

Comment:
I would have expected to see something like this here
```java
clientRepresentation.setProtocol("openid-connect");
```

but it seems to work without it for some reason anyway.